src/lib/other/checkboxdialog.cpp

```diff
@@ -16,47 +17,28 @@
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 * ============================================================ */
 #include "checkboxdialog.h"
-#include "ui_checkboxdialog.h"
+#include <QCheckBox>
 
-CheckBoxDialog::CheckBoxDialog(const QDialogButtonBox::StandardButtons &buttons, QWidget* parent)
-    : QDialog(parent, Qt::MSWindowsFixedSizeDialogHint | Qt::WindowTitleHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint)
-    , ui(new Ui::CheckBoxDialog)
+CheckBoxDialog::CheckBoxDialog(const QMessageBox::StandardButtons &buttons, QWidget* parent)
+    : QMessageBox(parent)
+    , m_checkBox(new QCheckBox)
 {
-    ui->setupUi(this);
-
-    ui->buttonBox->setStandardButtons(buttons);
-}
-
-void CheckBoxDialog::setIcon(const QIcon &icon)
-{
-    ui->iconLabel->setPixmap(icon.pixmap(48, 48));
-    ui->iconLabel->setFixedWidth(48);
-}
-
-void CheckBoxDialog::setText(const QString &text)
-{
-    ui->textLabel->setText(text);
+    setCheckBox(m_checkBox);
+    setStandardButtons(buttons);
+    setWindowFlags(windowFlags() | Qt::MSWindowsFixedSizeDialogHint | Qt::WindowTitleHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);
```

Comment:
Did you mean inheriting from QMessageBox or `setWindowFlags()`? As `setWindowFlags()` uses what was used in old `CheckBoxDialog` code.